collated/main/A0138411N.md

```diff
@@ -0,0 +1,877 @@
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(taskName, startDate, startTime, endDate, endTime, priority);
+    }
+
+}
+```
+###### /main/java/seedu/ggist/model/task/Task.java
+``` java
+    public Task(TaskName taskName, TaskDate startDate, TaskTime startTime, TaskDate endDate, TaskTime endTime, Priority priority) throws IllegalValueException {
+        this.taskName = taskName;
+        this.startDate = startDate;
+        this.startTime = startTime;
+        this.endDate = endDate;
+        this.endTime = endTime;
+        this.priority = priority;
+        if (startDate.value.equals(Messages.MESSAGE_NO_START_DATE_SPECIFIED) && startTime.value.equals(Messages.MESSAGE_NO_START_TIME_SET)) {
+            constructStartDateTime(endDate, endTime);
+        } else {
+            constructStartDateTime(startDate, startTime);
+        }
+        constructEndDateTime(endDate, endTime);
+        checkTimeClash();
+    }
+    
+
+    /**
+     * Copy constructor.
+     * @throws IllegalValueException 
+     */
+    public Task(ReadOnlyTask source) throws IllegalValueException {
+        this(source.getTaskName(), source.getStartDate(), source.getStartTime(), source.getEndDate(), source.getEndTime(), source.getPriority());
+    }
+    public void setDone() {
+        done = true;
+        setNotOverdue();
+    }
+    
+    public void setUndone() {
+        done = false;
+        if (end.before(new Date())) {
+            overdue = true;
+        }
+    }
+    
+    public void setNotOverdue() {
+        overdue = false;
+    }
+      
+    public void constructStartDateTime(TaskDate date, TaskTime time) throws IllegalValueException {
+        if ((date.value.equals(Messages.MESSAGE_NO_START_DATE_SPECIFIED) || date.value.equals(Messages.MESSAGE_NO_END_DATE_SPECIFIED)) && 
+            (time.value.equals(Messages.MESSAGE_NO_START_TIME_SET) || time.value.equals(Messages.MESSAGE_NO_END_TIME_SET))) {
+            Date date4 = new DateTimeParser("1st January 2050 11:59pm").getDateTime();
+            start =  date4;
+        } else if ((date.value.equals(Messages.MESSAGE_NO_START_DATE_SPECIFIED) || date.value.equals(Messages.MESSAGE_NO_END_DATE_SPECIFIED))){
+            Date date1 = new DateTimeParser("1st January 2050 " + time.value).getDateTime();
+            start =  date1;
+            System.out.println(start.toString());
+        } else if ((time.value.equals(Messages.MESSAGE_NO_START_TIME_SET) || time.value.equals(Messages.MESSAGE_NO_END_TIME_SET))) {
+            Date date2 = new DateTimeParser("11:59 pm " + date.value).getDateTime();
+            start =  date2;
+        } else {
+            Date date3 = new DateTimeParser(time.value + " " + date.value).getDateTime();
+            start =  date3;
+        }
+    }
+    
+    public void constructEndDateTime(TaskDate date, TaskTime time) throws IllegalValueException {
```

Comment:
much of the code in these few methods look similar; see if you can extract it out as a single method and call the method